### DISCUSSION TITLE
Pythoon / Pytest - Logging troubles

### DISCUSSION THREAD

**Discussion by TomKucera** [pytest-dev/pytest#10908] [Q&A]
Hi guys,

I have problem with logging in Pystest.

I have code, which produces some detail logs about processing. When I run this code in **Python**, all runs fine. But when test the same code using **PyTest**, the code itself is running, but without logging.

Where could be reason of this behaviour?

Thank you for help.

**Comment by arthuRHD** [comment]
Hi @TomKucera ! 👋🏻 

Let's dive a bit deeper, why do you need to show logs inside testing ?

### To ensure the code is working

launch pytest with argument `--capture=sys` to show your logs, or `--capture=no` for your `print()` calls and logs.
### To test the message within the logs

Capture the output. To capture the output, you can use this [fixture](https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture).


> I hope you found this comment helpful, if so feel free to validate my anwser. 👍🏻
